tagstudio/src/qt/ts_qt.py

```diff
@@ -948,6 +953,40 @@ def select_item(self, grid_index: int, append: bool, bridge: bool):
     def set_macro_menu_viability(self):
         self.autofill_action.setDisabled(not self.selected)
 
+    def update_completions_list(self, text: str) -> None:
+        matches = re.search(r"(mediatype|filetype|path|tag_id|tag):(\"?[A-Za-z0-9\ \t]+\"?)?", text)
+        if not matches:
+            return
+
+        query_type: str
+        query_value: str | None
+        query_type, query_value = matches.groups()
+
+        completion_list: list[str] = []
+
+        if query_type == "tag":
+            completion_list = list(map(lambda x: "tag:" + x.name, self.lib.tags))
```

Comment:
potentially dumb question: what is a tag's display name? its' shorthand?

Reply:
I've just realized that display names didn't make the jump from JSON... If #534 doesn't implement them then they'll need to be implemented in another new PR.

Essentially, they're a version of a Tag's named combined with their first parent tag in parenthesis (with preference to that parent tag's shorthand) to aide in telling tags with the same name apart. In 9.4 they were generated and grabbed from a [method](https://github.com/TagStudioDev/TagStudio/blob/140645943c331a0b752e5f09df7713936ce2b19b/tagstudio/src/core/library.py#L211) rather than being stored directly, which likely contributed to them going unnoticed during the migration so far.

Reply:
Tomorrow I'll look into getting the display names add in on https://github.com/TagStudioDev/TagStudio/pull/534, no point in having another PR just for that.